cli/plasmo/src/features/extension-devtools/project-creator.ts

```diff
@@ -0,0 +1,151 @@
+import spawnAsync from "@expo/spawn-async"
+import { copy, readJson, writeJson } from "fs-extra"
+import { writeFile } from "fs/promises"
+import { userInfo } from "os"
+import { resolve } from "path"
+import { temporaryDirectory } from "tempy"
+
+import { hasFlag, iLog, vLog } from "@plasmo/utils"
+
+import { generateGitIgnore } from "~features/extension-devtools/git-ignore"
+import {
+  PackageJSON,
+  generatePackage,
+  resolveWorkspaceToLatestSemver
+} from "~features/extension-devtools/package-file"
+import {
+  TemplatePath,
+  getTemplatePath
+} from "~features/extension-devtools/template-path"
+import type { PackageManagerInfo } from "~features/helpers/package-manager"
+
+import type { CommonPath } from "./common-path"
+
+const regex = /--with-[\w|-]+/
+
+export class ProjectCreator {
+  commonPath: CommonPath
+  templatePath: TemplatePath
+  packageManager: PackageManagerInfo
+  isExample = false
+
+  constructor(
+    _commonPath: CommonPath,
+    _packageManager: PackageManagerInfo,
+    _isExample = false
+  ) {
+    this.commonPath = _commonPath
+    this.packageManager = _packageManager
+    this.isExample = _isExample
+
+    this.templatePath = getTemplatePath()
+  }
+
+  async create() {
+    const withExampleName = process.argv.find((arg) => regex.test(arg))
+
+    return withExampleName !== undefined
+      ? this.createWith(withExampleName.substring(2))
+      : this.createBlank()
+  }
+
+  async createWith(exampleName: string) {
+    iLog(`Creating new project ${exampleName.split("-").join(" ")}`)
+
+    const { packageFilePath } = this.commonPath
+
+    // locate the tmp directory
+    const tempDirectory = temporaryDirectory()
+    vLog(`Download examples to temporary directory: ${tempDirectory}`)
+
+    // download the examples
+    await spawnAsync(
+      "git",
```

Comment:
We will need to add `git` as a required dependency in our docs

Reply:
Yes sire